sdk/python/kfp/components/_structures.py

```diff
@@ -65,12 +65,14 @@
 PrimitiveTypes = Union[str, int, float, bool]
 PrimitiveTypesIncludingNone = Optional[PrimitiveTypes]
 
+TypeType = Union[str, Dict, List]
```

Comment:
Should we rename this type? Would `ContainerType` or something similar be appropriate?

Reply:
This is the python type describing the type of the component input type.
So it's "[python] type of [input/output] type"

Reply:
Gotcha, would it make sense to indicate that? Such as `TypeIO` or `IOType`.

Reply:
Renamed it to `TypeSpecType`. It's also used for task arguments and some other properties...